.github/workflows/semantic-pr.yaml

```diff
@@ -0,0 +1,22 @@
+name: "Semantic Pull Request"
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+permissions:
+  pull-requests: write
```

Comment:
Can we throw a comment over this for why this is needed? Also, we shouldn't need write, only read here

Reply:
and there is a PR use this already such as #1837 so i added.

Reply:
I don't think we really need that feature IMO -- I'd be more in favor of not giving this workflow write access to PRs and just let folks set themselves in draft or out of draft as needed